__tests__/todo.spec.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react-native';
+import { build, fake } from '@jackfranklin/test-data-bot';
+import { firebase as mockFirebase } from '../src/firebase/config';
+
+import Todo from '../src/screens/todo';
+
+// mockFirebase.firestore().useEmulator('localhost', 8080);
+
+const userDataBuilder = build('User Data', {
+  fields: {
+    uid: fake((f) => f.random.uuid()),
+    fullName: fake((f) => f.name.findName()),
+    email: fake((f) => f.internet.email()),
+  },
+});
+const user = userDataBuilder();
+
+const itemBuilder = build('Fake Item', {
+  fields: {
+    key: fake((f) => f.random.uuid()),
+    word: fake((f) => f.lorem.word()),
+    sentence: fake((f) => f.lorem.sentence(3)),
+  },
+});
+const todoItem = itemBuilder();
+
+const submitInputText = (targetInput, itemText) => {
+  fireEvent.changeText(targetInput, itemText);
+  fireEvent(targetInput, 'submitEditing');
+};
+
+test('user is able to add an item to the to-do list', async () => {
+  //* render the todo screen
+  const { getByText, getByPlaceholderText, getByLabelText } = render(
+    <Todo user={user} />,
+  );
+
+  //* get to-do list and items counter
+  const counter = getByText(/Items: */i);
+  const todoList = getByLabelText(/todo-list/i);
+
+  //* type in the input field & click submits
+  submitInputText(
+    getByPlaceholderText(/What needs to be done?/i),
+    todoItem.word,
+  );
+
+  //* todo item should be added to the list and counter should increse by one
+  expect(todoList).toContainElement(getByText(todoItem.word));
+  expect(counter).toHaveTextContent(/ *1/i);
+});
+
+test('user should be able to toggle to-do item', async () => {
+  //* render the todo screen
+  const { getByPlaceholderText, getByRole, getByText, debug } = render(
+    <Todo user={user} />,
+  );
+
+  //* type in the input field & click submits
+  submitInputText(
+    getByPlaceholderText(/What needs to be done?/i),
+    todoItem.word,
+  );
+
+  //* get to-do item text and switch
+  const itemText = getByText(todoItem.word);
+
+  //* toggle switch
+  fireEvent(getByRole('switch'), 'onValueChange', true);
```

Comment:
**Though.** if other dev look at this line, it will feel this is part of the way to test, when actually the value is the next line. I feel we should remove it, don't you?

Reply:
I dont get what are you saying, remove `const itemText = getByText(todoItem.word)`, or something else? i could do `expect(getByText(todoItem.word)).toHaveStyle([{ textDecorationLine: 'line-through' }])`

Reply:
My bad I read wrongly.